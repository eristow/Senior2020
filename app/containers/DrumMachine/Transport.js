import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Wrapper = styled.div`
  display: flex;
  flex: 0 1 auto;
  flex-wrap: wrap;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;
`;

export default function Transport({ children }) {
  return <Wrapper>{children}</Wrapper>;
}

Transport.propTypes = {
  children: PropTypes.array,
};
