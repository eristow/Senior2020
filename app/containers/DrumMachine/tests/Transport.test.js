import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import configureStore from '../../../configureStore';

import Transport from '../Transport';

describe('<Transport />', () => {
  let store;

  beforeAll(() => {
    store = configureStore();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();

    render(
      <Provider store={store}>
        <Transport dispatch={dispatch} />
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <Transport />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
