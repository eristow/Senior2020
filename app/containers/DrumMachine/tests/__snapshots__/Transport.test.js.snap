// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Transport /> Should render and match the snapshot 1`] = `
<div
  class="Transport__Wrapper-sc-16zg5lv-0 qZwmM"
/>
`;
